### PR TITLE
[homework#6]-JS-BASICS by Ilya Ischenko

### PR DESCRIPTION
**Pull request by ilya07092000**

### CODE REVIEW

File: homeworks/ilya-ischenko/06-js-basic/script.js [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+  if (even(s.length)) {
+    return s.slice(middle - 1, middle + 1);
+  }
+
+  return s.slice(middle - 1, middle);
+};
+console.log(getMiddle('testing'));
+
+// 7 - http://www.codewars.com/kata/partition-on
+function partitionOn (pred, items) {
```
[the remark this answers is not in the document]
Comment (ilya07092000) [reply homeworks/ilya-ischenko/06-js-basic/script.js:115]: Понял, поправил решение

File: homeworks/ilya-ischenko/06-js-basic/script.js [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+      return tempArr[0];
+    }
+  }
+
+  return [];
+};
+console.log(findOdd([20, 1, -1, 2, -2, 3, 3, 5, 5, 1, 2, 4, 20, 4, -1, -2, 5]));
+
+// 10 - https://www.codewars.com/kata/find-the-parity-outlier
+const findOutlier = function(integers) {
```
Comment (esvyridov) [inline homeworks/ilya-ischenko/06-js-basic/script.js:160]: I would say that if there are always 3 elements in the array, we can identify the type (odd or even) of number which we are looking for based on these 3 first elements. That can decrease the number of iterations and memory usage. That's just FYI, you can try to implement this if you want :)

File: homeworks/ilya-ischenko/06-js-basic/script.js [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+    }
+  };
+
+  return result.join('');
+};
+console.log(stringBreakers(5, 'This is an example string'));
+
+// 20 - https://www.codewars.com/kata/514a024011ea4fb54200004b
+const domainName = function(url) {
+  let copyUrl = url;
```
Comment (esvyridov) [inline homeworks/ilya-ischenko/06-js-basic/script.js:362]: I would chain the `replace` methods here

**Review by esvyridov — APPROVED** [review]

File: homeworks/ilya-ischenko/06-js-basic/script.js (hunk 2 above)
Comment (ilya07092000) [reply homeworks/ilya-ischenko/06-js-basic/script.js:160]: Maybe I don't understand you, but we are not always have 3 elements, it can be more than 3.

Comment (ilya07092000) [reply homeworks/ilya-ischenko/06-js-basic/script.js:160]: Maybe I don't understand you, but we are not always have 3 elements, it can be more than 3.

Comment (esvyridov) [reply homeworks/ilya-ischenko/06-js-basic/script.js:160]: Yes, but there are at least 3 elements. Based on them we can identify what type of number we are looking for. For example:
[1, 2, 3, ...] - we are looking for an even number
[1, 3, 5, ...] - we are looking for an even number.

Другими словами - основываясь на первый трёх числах массива мы можем определить какой тип (чётное, нечётное) числа приобладает во всём массиве. После того как эта информация получена - можно начинать искать этот тип со сложностью О(n)

Comment (ilya07092000) [reply homeworks/ilya-ischenko/06-js-basic/script.js:160]: Теперь понял, сейчас поправлю)

**Review by esvyridov — APPROVED** [review]

**Pull request merged**